image-processing/linefinder.py

```diff
@@ -1,183 +1,51 @@
-
-          '''
-        
-        x = linefinder.blur_sample_gauss(self, False)
-        peak_positions = find_peaks(x[self.row])
-
-        if view_plot == True:
-
-            fig, ax = plt.subplots(ncols=3, nrows=1)
-            fig.suptitle('Detecting lines through Scipy find_peaks')
-
-            ax[0].imshow(x, cmap='gray')
-            ax[0].set(xlabel='', ylabel='', title = r'Blurred Sample, $\sigma$ = {}'.format(self.sigma))
-
-            ax[1].plot(np.arange(0,np.size(x[self.row]), 1), x[self.row])
-            ax[1].set(xlabel='', ylabel='', title = 'Values along row {} of blurred sample'.format(self.row))
-
-            ax[2].imshow(self.original, cmap='gray')
-            ax[2].vlines(x=peak_positions[0], color = 'red', ymin=0, ymax=len(x), linewidth=1)
-            ax[2].set(xlabel='', ylabel='', title = 'Detected lines')
-
-            plt.show()
-
-        return peak_positions[0]
-
-
-
-    def find_prominences(self, view_plot = True):
-        '''
-        Finds the prominence of the peaks found by the scipy_peaks method.
-
-            INPUTS:
-            self
-            view_plot - set True to view the output plot 
-
-            OUTPUTS:
-            fig - figure showing the blurred sample, the pixel values along the chosen rows, and the pixel value with marked peaks and lines drawn to show prominence 
-
-        '''
-        
-        x = linefinder.blur_sample_gauss(self, False)
-        y = linefinder.scipy_peaks(self, False)
-        row_looking_at = x[self.row]
-        prominences = peak_prominences(row_looking_at,y)[0] #this zero is needed to return the array of the prominences, excluding extra information
-        
-        if view_plot == True: 
-
-            heights = row_looking_at[y] - prominences
-            fig, ax = plt.subplots(ncols=3, nrows=1)
-            fig.suptitle('Finding Peak Prominences')
-
-            ax[0].imshow(x, cmap='gray')
-            ax[0].set(xlabel='', ylabel='', title = r'Blurred Sample, $\sigma$ = {}'.format(self.sigma))
-
-            ax[1].plot(np.arange(0,np.size(row_looking_at), 1), row_looking_at)
-            ax[1].set(xlabel='', ylabel='', title = 'Values along row {} of blurred sample'.format(self.row))
-
-            ax[2].plot(np.arange(0,np.size(x[self.row]), 1), x[self.row])
-            ax[2].plot(y, row_looking_at[y], "x")
-            ax[2].vlines(x=y, color = 'red', ymin=heights, ymax=row_looking_at[y], linewidth=1)
-            ax[2].set(xlabel='', ylabel='', title = 'Peak Prominences')
-
-            plt.show()
-
-        return prominences
-
-
-
-    def find_lines_with_exclusions(self, view_plot = True, distance = False, min_prominences = False):
+        self.gauss_blur = ndimage.gaussian_filter(self.original, self.sigma)
+        self.peak_positions = [find_peaks(row)[0] for row in self.gauss_blur]  #used list comprehension to turn from a normal list of peak positions of one row, to a list of lists of peak positions of every row
```

Comment:
this is an awesome addition (rip my future work section) but what's it like for runtime? 
We're essentially now running the previous code about 537 times, if I'm remembering the number of rows right

Reply:
runtime looking good!! since it's just dealing with numbers it usually isn't an issue unless your RAM is maxxed out